content/en/blog/_posts/2022-11-21-devicemanager-ga.md/index.md

```diff
@@ -0,0 +1,95 @@
+
+### Introduction
+Device Plugin framework was introduced in Kubernetes v1.8 release as a vendor
+independent framework to enable discovery, advertisement and allocation of external
+devices without modifying core Kubernetes. The feature graduated to Beta in v1.10.
+Due to its widespread use and adoption, this feature is being graduated to GA in
+v1.26.
+
+Device Manager in Kubelet facilitates communication with device plugins over gRPC
+through Unix sockets. Device Manager and Device plugins both act as gRPC servers
+and clients by serving and connecting to the exposed gRPC services respectively.
+Device plugins serve a gRPC service that Kubelet connects to for device discovery,
+advertisement (as extended resources) and allocation. Device Manager connects to
+the `Registration` gRPC service served by Kubelet to register itself with Kubelet.
+
+### Pod requesting a device
+Please refer to the documentation for an [example](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#example-pod) on how a pod can request a device exposed to the cluster by a device plugin.
+
+### Example Device Plugin implmentations 
+Here are some device plugins for example implmentations:
+- [NVIDIA device plugin for Kubernetes](https://github.com/NVIDIA/k8s-device-plugin)
+- [Collection of Intel device plugins for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes)
+- [SRIOV network device plugin for Kubernetes](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin)
+- [AMD GPU device plugin](https://github.com/RadeonOpenCompute/k8s-device-plugin)
+
+## Noteworthy developments since Device Plugin framework introduction
+
+### Kubelet APIs moved to kubelet staging repo
+External facing `deviceplugin` API packages moved from `k8s.io/kubernetes/pkg/kubelet/apis/`
+to `k8s.io/kubelet/pkg/apis/` in v1.17. Refer to [Move external facing kubelet apis to staging](https://github.com/kubernetes/kubernetes/pull/83551) for more details on the rationale behind this change.
+
+### Device Plugin API updates
+Additional gRPC endpoints introduced:
+  1. `GetDevicePluginOptions` is used by device plugins to communicate
+     options to the `DeviceManager` in order to indicate if `PreStartContainer`
+     or `GetPreferredAllocation` calls are supported and can be called before
+     making devices available to the container.
+  1. `GetPreferredAllocation` allows a device plugin to forward allocation
+     preferrence to the `DeviceManager` so it can incorporate this information
+     into its allocation decisions. The `DeviceManager` will call out to a
+     plugin at pod admission time asking for a preferred device allocation
+     of a given size from a list of available devices to make a more informed
+     decision.
+  1. `PreStartContainer` is called before each container start if indicated by
+     device plugins during registration phase. It allows Device Plugins to run device
+     specific operations on the Devices requested.
```

Comment:
we should have examples of this specific operations (if memory serves, device reset?). If we have easily available examples, e.g. in the KEP or doc, we should probably list some them here.

Reply:
Sure, we can mention reconfiguring or reprogramming FPGAs before the container starts as a use case here.